### ISSUE TITLE
Speed up filtering step

### ISSUE THREAD

**Issue by samlhao** [czbiohub/sc2-illumina-pipeline#50]
The host read filtering + Kraken2 classification is the bottleneck right now. OpenGene (who made fastp) recently came out with their [new classification tool for SC2](https://github.com/OpenGene/fastv), which seems to be faster than the two separate steps and can output clean reads for upload to SRA. I haven't done extensive testing, but this could be worth looking into once we have some time - hopefully it wouldn't be too hard to do a comparison with the benchmarks @jackkamm and @neevor made.

**Comment by neevor** [comment]
It sounds like this tool would filter out any k-mers that don't map to an input fasta. This can (and there are pre-built ones) include common virus and microorganisms found in humans. This is a bit different than just filtering out human but seems fine to me unless we are trying to capture some very exotic stuff. Just wanted to verify that this still works for what we need.

**Comment by samlhao** [comment]
That would match what the kraken2 step is doing currently, so I think it's fine. For studies where we might care about coinfections, I imagine we'd just use the IDSeq output.

**Comment by neevor** [comment]
Ok sounds good. I just wanted to make sure since we will no longer have a step that produced raw reads with just the host removed but sounds like that's fine.

**Comment by samlhao** [comment]
I guess depending on how much speed up we get, we could make it an option to do human genome alignment upfront, or even just keep it for now. I'm not too sure `fastv` is exactly the tool we want replacing everything, but it might give us a speed up over kraken2 + trim_galore, since it's got `fastp` built in. I'll defer to @jackkamm on whether or not we want to keep the human genome alignment step.

**Comment by jackkamm** [comment]
Currently we have 2 steps:
- Filter host reads
- Filter non-SARS-CoV-2 reads
I made the host-filtering step optional because it can be very slow and is not always needed. It is primarily needed if we want to upload metagenomic fastqs that include non-SARS-CoV-2 reads. Otherwise, if we are fine with just uploading the SARS-CoV-2 reads, then the second step is sufficient.

It's true we can use IDseq to filter host instead, that's what we did for the upload for the host-expression paper. Nice to have it built into the pipeline but not absolutely critical.

Maybe we can make the host-filtering step off by default, since we are less commonly doing metagenomics these days.

I think both these steps, when they're on, should publish their respective filtered fastq files to S3.
